### ISSUE TITLE
[BUG] Cannot use entity files or create entity files from imported mappings files

### ISSUE THREAD

**Issue by jaccoTopVintage** [laravel-doctrine/orm#143]
### Expected behaviour

```
$> php artisan doctrine:generate:entities /

Generating getter and setter for default entity manager...
Processing entity "App\Entities\Nnnnnnnn"

Entity classes generated to "D:\Web\TopManager"
```
### Actual behaviour

```
$> php artisan doctrine:generate:entities /

Generating getter and setter for default entity manager...

  [Doctrine\Common\Persistence\Mapping\MappingException]
  Invalid mapping file 'App.Entities.Nnnnnnnn.orm.xml' for class 'App\Entities\Nnnnnnnn'.
```
### Steps to reproduce the behaviour
- Setup a new Laravel project.
- In the configuration file `config/doctrine.php` make sure the following lines are set as shown below:

```
'managers' => [
    'default' => [
        'meta' => 'SimplifiedXml',
        'paths' => [
            base_path('database/mappings') => 'App\Entities',
        ],
    ],
];
```
- Create the following additional file structure:

```
[project]
   +- [App]
       +- [Entites]
   +- [database]
       +- [mappings]
```
- create mappings files from an existing database using console command:  
  `php artisan doctrine:mapping:import xml database/mappings`
- attempt to generate entities from the imported mappings files using console command:  
  `php artisan doctrine:generate:entities /`

**Comment by jaccoTopVintage** [comment]
## Additional information

I think there needs to be a change to a Doctrine class to accommodate the use of mapping files (`.xml`/`.yml`) for Doctrine when used in combination with Laravel.

More specifically, in file [`Doctrine/Common/Persistence/Mapping/Driver/FileDriver.php`](https://github.com/doctrine/common/blob/v2.6.1/lib/Doctrine/Common/Persistence/Mapping/Driver/FileDriver.php):

**Current lines of code**

```
116: if (!isset($result[$className])) {
117:     throw MappingException::invalidMappingFile($className, str_replace('\\', '.', $className) . $this->locator->getFileExtension());
118: }
119:
120: return $result[$className];
```

**Required lines of code**

```
116: $classNameBase = explode('\\', $className);
117: $classNameBase = end($classNameBase);
118: if (!isset($result[$classNameBase])) {
119:     throw MappingException::invalidMappingFile($className, $classNameBase . $this->locator->getFileExtension());
120: }
121: return $result[$classNameBase];
```

<br>
I would be happy to contribute this change to with a pull request, but I need some help:

Could you guys point me in the right direction on: how to overwrite [the class method](https://github.com/doctrine/common/blob/v2.6.1/lib/Doctrine/Common/Persistence/Mapping/Driver/FileDriver.php) mentioned above, which is part of [doctrine2/common](https://github.com/doctrine/common), in such a way that the overwritten behaviour can be bundled as part of [laravel-doctrine/orm](https://github.com/laravel-doctrine/orm)?

**Comment by jaccoTopVintage** [comment]
## Additional Issue

In addition to the issue listed in the OP, the change as described in the 2nd post is also required to _use_ the entity files.

**In other words, without the proposed change, it is seems to be impossible to use `xml` or `yaml` files with Laravel.**

If there turns out to be a some way of using mapping files without the changes to the Doctrine/common file, I would be very happy. However so far, all the things we've tried all came back to the same issue, requiring the change as described.

**Comment by jaccoTopVintage** [comment]
Finally found the solution to this issue:

It seems the automatically generated mapping files need more manual editing before the can be used.
The `name` attribute for the `entity` tag in the XML mappings file needs to be prepended with the namespace:

**autogenerated**
`<entity name="Nnnnnnnnnnnnnn" table="Nnnnnnnnnnnnnns">`

**required**
`<entity name="App\Entities\Nnnnnnnnnnnnnn" table="Nnnnnnnnnnnnnns">`

<br>
In summary, **this issue can be closed**; the suggested changes are not needed.

However, the required manual editing may very well point to an issue with the automatically generated xml files.

**Comment by guiwoda** [comment]
@jaccoTopVintage there is a `namespace` configuration key. Would you mind testing your xmls _without_ the namespace, and your `doctrine.php` file with them instead?

```
'paths' => [
    base_path('database/mappings'),
],
'namespaces' => [
    'App\\Entities',
],
```

**Comment by jaccoTopVintage** [comment]
just dropping a note to mention I've not forgotten this issue and will try adding the `namespaces` config key, as soon as I've time to do so (but: quite busy at the moment).

**Comment by jaccoTopVintage** [comment]
Ok,

I've tried your suggestion, but to no avail. In fact, I've lots of combinations for the configuration settings, but none give a working result (see bottom of post for log).

The easiest method I found to make the error go away is to manually edit each created mapping file. But since more than 100 tables, it is not very convenient so to say.

The file as created by the command `php artisan doctrine:mapping:import xml database/mappings`:

``` xml
<?xml version="1.0" encoding="utf-8"?>
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
  <entity name="Users" table="users">
    (...)
  </entity>
</doctrine-mapping>
```

Edit the third line to include the `namespace` in the entity name

``` xml
<?xml version="1.0" encoding="utf-8"?>
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
  <entity name="App\Entities\Users" table="users">
    (...)
  </entity>
</doctrine-mapping>
```

<br><br>

<hr>

<br><br>
**Passing the namespace as an argument**

I also tried feeding the namespace as an argument to the mappings import command: `php artisan doctrine:mapping:import xml database/mappings --namespace=App/Entities/`, but this does _not_ give the desired effect:

The mapping is now written to a file named `App.Entities.Users.orm.xml`. The contents of the file looks good (entity name has the namespace prepended):

``` xml
<?xml version="1.0" encoding="utf-8"?>
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
  <entity name="App\Entities\Users" table="users">
    (...)
  </entity>
</doctrine-mapping>
```

But, when I try to generate my entities using command `php artisan doctrine:generate:entities /` I get:

```
[Doctrine\Common\Persistence\Mapping\MappingException]
Invalid mapping file 'App.Entities.App.Entities.Users.orm.xml' for class 'App\Entities\App\Entities\Users'.
```

<br><br>

<hr>

<br><br>

**Log of various configuration settings combinations**

Importing mappings with:
`php artisan doctrine:mapping:import xml database/mappings`

Generating entities with (the errors are output from this command):
`php artisan doctrine:generate:entities /`

```
---------------------------------------------------------------
Settings:
    'namespaces' => [
        'App\Entities'
    ],
    'paths'      => [
        base_path('database/mappings') => 'App\Entities',
    ],
Result:
    File:       Users.orm.xml
    Contents:   <entity name="Users" table="users">
Error:
    [Doctrine\Common\Persistence\Mapping\MappingException]
    Invalid mapping file 'App.Entities.Users.orm.xml' for class 'App\Entities\Users'.
---------------------------------------------------------------
Settings:
    'namespaces' => [],
    'paths'      => [
        base_path('database/mappings') => 'App\Entities',
    ],
Result:
    File:       Users.orm.xml
    Contents:   <entity name="Users" table="users">

Error:
    [Doctrine\Common\Persistence\Mapping\MappingException]
    Invalid mapping file 'App.Entities.Users.orm.xml' for class 'App\Entities\Users'.
---------------------------------------------------------------
Settings:
    'namespaces' => [
        'App\Entities'
    ],
    'paths'      => [
        base_path('database/mappings'),
    ],
Result:
    File:       Users.orm.xml
    Contents:   <entity name="Users" table="users">
Error:
    [Doctrine\Common\Persistence\Mapping\MappingException]
    File mapping drivers must have a valid directory path, however the given path 0 seems to be incorrect!
---------------------------------------------------------------
Settings:
    'namespaces' => [
        'App\Entities'
    ],
    'paths'      => [
        'App\Entities',
    ],
Result:
    File:       Users.orm.xml
    Contents:   <entity name="Users" table="users">
Error:
    [Doctrine\Common\Persistence\Mapping\MappingException]
    File mapping drivers must have a valid directory path, however the given path 0 seems to be incorrect!
---------------------------------------------------------------
Settings:
    'namespaces' => ['\\'],
    'paths'      => [
        base_path('database/mappings') => 'App\Entities',
    ],
Result:
    File:       Users.orm.xml
    Contents:   <entity name="Users" table="users">

Error:
    [Doctrine\Common\Persistence\Mapping\MappingException]
    Invalid mapping file 'App.Entities.Users.orm.xml' for class 'App\Entities\Users'.
---------------------------------------------------------------
Settings:
    'namespaces' => [
        'App\Entities'
    ],
    'paths'      => [
        base_path('database/mappings') => '',
    ],
Result:
    File:       Users.orm.xml
    Contents:   <entity name="Users" table="users">
Error:
    [Doctrine\Common\Persistence\Mapping\MappingException]
    Invalid mapping file '.Users.orm.xml' for class '\Users'.
---------------------------------------------------------------
Settings:
    'namespaces' => [
        'App\Entities'
    ],
    'paths'      => [
        base_path('database/mappings') => '\\',
    ],
Result:
    File:       Users.orm.xml
    Contents:   <entity name="Users" table="users">
Error:
    [Doctrine\Common\Persistence\Mapping\MappingException]
    Invalid mapping file '..Users.orm.xml' for class '\\Users'.
```

**Comment by Jacco-V** [comment]
Ran into the same issue

The problem seems to be twofold:
1. Laravel-Doctrine ignores the `namespaces` directive in the `/config/doctrine.php` config file.
2. When the `--namespace=` command line argument is specified (as described by @jaccoTopVintage <sup><sub>(nice name btw)</sub></sup> above) the namespace is both prepended to both to the file name (_wrong_) and the entity name in the XML file contents (_correct_).

<br>
So: **[BUG]**

### Package version, Laravel version

Package v1.1.10,
Laravel v5.2.

### Expected behaviour

Using the `/config/doctrine.php` configuration (exactly the same as the default):

```
return [
    'managers' => [
        'default' => [
            'dev'        => env('APP_DEBUG'),
            'meta'       => env('DOCTRINE_METADATA', 'SimplifiedXml'),
            'connection' => env('DB_CONNECTION_DEFAULT', 'default'),
            'namespaces' => [
                'App\Entities'
            ],
            'paths'      => [
                base_path('database/mappings') => 'App\Entities',
            ],
            'repository' => Doctrine\ORM\EntityRepository::class,
            'proxies'    => [
                'namespace'     => false,
                'path'          => storage_path('proxies'),
                'auto_generate' => env('DOCTRINE_PROXY_AUTOGENERATE', false)
            ],
            'events'     => [
                'listeners'   => [],
                'subscribers' => []
            ],
            'filters'    => [],
            'mapping_types'              => [
                'enum'   => 'string'
            ]
        ],
    ],
];
```

When using command `php artisan doctrine:mapping:import xml` 

My mapping file will be written to path `/database/mappings` (as specified by the config file).

My mapping file is named:  `User.orm.xml`<sup>1</sup> (note the singular) and the contents of this XML file reads:

```
<?xml version="1.0" encoding="utf-8"?>
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
  <entity name="App\Entities\User" table="users">
    <!-- entity data -->
  </entity>
</doctrine-mapping>
```

Note the namespace in the `<entity name="...">` element attribute..

### Actual behaviour

Using the exact same configuration as above.
When using command `php artisan doctrine:mapping:import xml` (same as above)

My mapping file will be written to path `/app/Mappings`

My mapping file is named:  `Users.orm.xml` (note the plural) and the contents of this XML file reads:

```
<?xml version="1.0" encoding="utf-8"?>
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
  <entity name="Users" table="users">
    <!-- entity data -->
  </entity>
</doctrine-mapping>
```

Note the _missing_ namespace in the `<entity name="...">` element attribute.

### Steps to reproduce the behaviour

Use the command described above

<br>
<sup>1</sup>: For clarity, I've used the table <-> entity: `users` <-> `User` in this whole bug report.

**Comment by patrickbrouwers** [comment]
Just tested this myself:

`php artisan doctrine:mapping:import xml database/mappings --namespace=App\\Entities\\`

(Note the backwards slashes)

In `database/mappings/App.Entities.Table.orm.xml`:

```
<?xml version="1.0" encoding="utf-8"?>
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
  <entity name="App\Entities\Table" table="table">
    <id name="id" type="integer" column="id">
      <generator strategy="IDENTITY"/>
    </id>
  </entity>
</doctrine-mapping>
```

Generating entities only works if I change `.orm` to `.dcm`: entity got created with namespace.  It seems the original Symfony command doesn't have `dcm` as default. For `SimplifiedXml` this is the default (.orm). 

It seems you are trying to generate `xml` and treating it as `SimplifiedXml`. The Xml driver has the the notation like this: `App.Entities.Table.dcm.xml`. (While the simplified version only uses the class name) 
So the import is completely correct. Doctrine doesn't support SimplifiedXml mapping import, so without modifying the filename, this will not work.

SimplifiedXml import added on our side: https://github.com/laravel-doctrine/orm/commit/d2b647a3b95c243c01755a5027585e4f1238f532
